tests/basic/nomad_test.go

```diff
@@ -0,0 +1,415 @@
+package tests
+
+import (
+	"fmt"
+	"github.com/hashicorp/nomad/api"
+	. "github.com/onsi/ginkgo/v2"
+	"github.com/portworx/torpedo/drivers/nomad"
+	"github.com/portworx/torpedo/pkg/log"
+	. "github.com/portworx/torpedo/tests"
+	"time"
+)
+
+var _ = Describe("{ListNodesOnNomadCluster}", func() {
+	var client *nomad.NomadClient
+	var err error
+	JustBeforeEach(func() {
+		StartTorpedoTest("ListNodesOnNomadCluster", "List nodes in Nomad Cluster", nil, 0)
+		client, err = nomad.NewNomadClient()
+		log.FailOnError(err, "Failed to get Nomad Client")
+	})
+	It("List and print Nodes", func() {
+		nodes, err := client.ListNodes()
+		log.FailOnError(err, "Failed to list nodes")
+		for _, node := range nodes {
+			log.Infof("Nodename present : %v", node.Name)
+		}
+	})
+})
+
+var _ = Describe("{CreateAndValidateNomadVolume}", func() {
+	var client *nomad.NomadClient
+	var err error
+	var volumes []*api.CSIVolume
+
+	BeforeEach(func() {
+		StartTorpedoTest("CreateAndValidateNomadVolume", "Create and validate a volume in Nomad", nil, 0)
+		client, err = nomad.NewNomadClient()
+		log.FailOnError(err, "Failed to get Nomad Client")
+	})
+
+	It("Should create a volume and validate it's active", func() {
+		volumeID := "test-volume"
+		pluginID := "portworx"
+		capacityMin := int64(1 * 1024 * 1024 * 1024)
+		capacityMax := capacityMin
+		accessMode := "single-node-writer"
+		attachmentMode := "file-system"
+		fsType := "ext4"
+		mountFlags := []string{}
+
+		err := client.CreateVolume(volumeID, pluginID, capacityMin, capacityMax, accessMode, attachmentMode, fsType, mountFlags)
+		log.FailOnError(err, "Failed to create volume")
+
+		// Validate the volume
+		volume, err := client.VolumeInfo(volumeID)
+		log.FailOnError(err, "Failed to get volume info")
+		volumes = append(volumes, volume)
+		if volume.ID == volumeID {
+			log.Infof("Volume %v created and validated successfully", volumeID)
+		} else {
+			log.Errorf(" Volume %v not created successfully")
+		}
+	})
+	JustAfterEach(func() {
+		for _, volume := range volumes {
+			err = client.DeleteVolume(volume.ID)
+			if err != nil {
+				log.Errorf("Volume %v could not be deleted with err %v", volume.ID, err)
```

Comment:
dont you want to use failonerror?

Reply:
Yes - I am changing the cleanup logic in my next PR. This one by one logic will be moved to loop based.